mmcls/apis/inference.py

```diff
@@ -1,6 +1,105 @@
-def init_model():
-    pass
+import warnings
 
+import mmcv
+import numpy as np
+import torch
+from mmcv.parallel import collate, scatter
+from mmcv.runner import load_checkpoint
 
-def inference_model():
-    pass
+from mmcls.core import get_classes
+from mmcls.datasets.pipelines import Compose
+from mmcls.models import build_classifier
+
+
+def init_model(config, checkpoint=None, device='cuda:0'):
+    """Initialize a classifier from config file.
+
+    Args:
+        config (str or :obj:`mmcv.Config`): Config file path or the config
+            object.
+        checkpoint (str, optional): Checkpoint path. If left as None, the model
+            will not load any weights.
+
+    Returns:
+        nn.Module: The constructed classifier.
+    """
+    if isinstance(config, str):
+        config = mmcv.Config.fromfile(config)
+    elif not isinstance(config, mmcv.Config):
+        raise TypeError('config must be a filename or Config object, '
+                        f'but got {type(config)}')
+    config.model.pretrained = None
+    model = build_classifier(config.model)
+    if checkpoint is not None:
+        map_loc = 'cpu' if device == 'cpu' else None
+        checkpoint = load_checkpoint(model, checkpoint, map_location=map_loc)
+        if 'CLASSES' in checkpoint['meta']:
+            model.CLASSES = checkpoint['meta']['CLASSES']
+        else:
+            warnings.simplefilter('once')
+            warnings.warn('Class names are not saved in the checkpoint\'s '
+                          'meta data, use imagenet by default.')
+            model.CLASSES = get_classes('imagenet')
+    model.cfg = config  # save the config in the model for convenience
+    model.to(device)
+    model.eval()
+    return model
+
+
+class LoadImage(object):
```

Comment:
Can we use the test pipeline instead?

Reply:
Ref: https://github.com/open-mmlab/mmdetection/blob/master/mmdet/apis/inference.py#L78

Reply:
It has been modified accordingly.

Reply:
I found some errors when a try a workflow like that: workflow = [('train', 2),('val', 1)] 
I got an error that 'gt_label' is unxpected attribute. This is because single_test_gpu has the parameter returnLoss=False fixed